src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -949,7 +960,15 @@ private bool ParseFile(string[] args, ref int i, bool noexitSeen)
                             }
                             else
                             {
-                                _collectedArgs.Add(new CommandParameter(arg.Substring(0, offset), arg.Substring(offset + 1)));
+                                bool? boolValue = GetBoolValue(arg.Substring(offset + 1));
```

Comment:
null-coalescing operator requires same type on both sides, I can use ternary operator to simplify it instead